logic/src/commonMain/kotlin/com/wire/kalium/logic/data/message/draft/MessageDraft.kt

```diff
@@ -0,0 +1,27 @@
+/*
+ * Wire
+ * Copyright (C) 2024 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+package com.wire.kalium.logic.data.message.draft
+
+import com.wire.kalium.logic.data.message.mention.MessageMention
+
+data class MessageDraft(
```

Comment:
What if instead of adding nullables, we create sealed implementations?

`DraftMessage.Reply`

`DraftMessage.Quote`

`DraftMessage.Simple`

Reply:
I think it's not worth it because it can be divided only into 2 types which one doesn't have quoted Id and one which has. Rest variables can be combined (edited message can have quote, mentions) and at the end it will be merged into one entity object saved in db

Reply:
Indeed. I forgot about the possibility of it being both Reply and Edit at the same time.